rust/lance-index/src/vector/residual.rs

```diff
@@ -77,15 +78,12 @@ where
     });
 
     let residuals = vectors_slice
-        .par_chunks(dimension)
+        .chunks_exact(dimension)
```

Comment:
we need to get this back after fixed #2503 right?

Reply:
I don't think so.  I don't think this parallelism is helpful.

Reply:
At least, not on the training path.  If we do this transform on the query path then it might be useful.  Even then, if we have many queries per second (many concurrent queries) I think we might be better without it.